taskcluster/client.py

```diff
@@ -535,9 +535,7 @@ def _makeHttpRequest(self, method, route, payload):
                 )
 
             # Try to load JSON
-            try:
-                return response.json()
-            except ValueError:
+            if response.headers['Content-Type'].strip() != 'application/json':
```

Comment:
This is no longer parsing the body if it is `application/json`, which we need to happen.  We also still need to put this in a try/catch.